### DISCUSSION TITLE
How to freeze layers / heads in Faster RCNN?

### DISCUSSION THREAD

**Discussion by yribeiro** [open-mmlab/mmdetection#9282] [Q&A]
I am trying to train the Faster RCNN network. I have used a pretrained model using the load_from config var. As the paper outlines, we need to train the RPN first, freeze the RPN network head and train the FCN classifier on top. 

Is there an easy way to freeze components, so that after I am done training the RPN, I can freeze the module and only train the FCN classifier on top?

My initial guess was that I would need to inherit the original FasterRCNN detector and extend / modify the bits needed (loss / head etc). Is there an easier way to do this? Via configs maybe?

**Comment by Kegard** [comment]
i also have this question. ann my guess is that add the "freeeze" function in the RPN and freeze it by modify the configs. but i don't know whether it is ok.

**Comment by josh3255** [comment]
@yribeiro @Kegard  you can freeze backbone using
```python
backbone = model.backbone
for param in backbone.parameters():
    param.requires_grad = False
```

**Reply by Kegard** [reply]
Thank you for your reply, I want to know where to modify the code?  I think  just modify the configs is invalid. Should i modify the code in the mmdet/models/backbones/resnet.py?

**Reply by josh3255** [reply]
@Kegard You can add the code after loading the model from train.py or test.py

**Reply by Kegard** [reply]
Thank you!  I get it
